content/source/docs/extend/testing/acceptance-tests/teststep.html.md

```diff
@@ -0,0 +1,291 @@
+documentation on godoc.org:
+
+- [TestCheckResourceAttrSet(name, key
+string)](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckResourceAttrSet)  
+- [TestCheckModuleResourceAttrSet(mp []string, name string, key
+string)](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckModuleResourceAttrSet)  
+- [TestCheckResourceAttr(name, key, value
+string)](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckResourceAttr)  
+- [TestCheckModuleResourceAttr(mp []string, name string, key string, value
+string)](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckModuleResourceAttr)
+- [TestCheckNoResourceAttr(name, key
+string)](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckNoResourceAttr)  
+- [TestCheckModuleNoResourceAttr(mp []string, name string, key
+string)](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckModuleNoResourceAttr)  
+- [TestCheckResourceAttrPtr(name string, key string, value
+*string)](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckResourceAttrPtr)  
+- [TestCheckModuleResourceAttrPtr(mp []string, name string, key string, value
+*string)](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckModuleResourceAttrPtr)  
+- [TestCheckResourceAttrPair(nameFirst, keyFirst, nameSecond, keySecond
+string)](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckResourceAttrPair)  
+- [TestCheckModuleResourceAttrPair(mpFirst []string, nameFirst string, keyFirst
+string, mpSecond []string, nameSecond string, keySecond
+string)](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckModuleResourceAttrPair)  
+- [TestCheckOutput(name, value
+string)](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckOutput)  
+
+
+## Custom check functions 
+
+The `Check` field of `TestStep` accepts any function of type
+[TestCheckFunc](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckFunc).
+Developers are free to write their own `check` functions to create customized
+validation functions for their plugin. Any function that matches the
+`TestCheckFunc` function signature of `func(*terraform.State) error` can be used
+individually, or with other `TestCheckFunc` functions with one of the above
+Aggregate functions. 
+
+It's common to write custom `TestCheckFunc` functions to validate resources were
+created correctly by using SDKs directly to verify identity and properties of
+resources. These functions can retrieve information by SDKs and provide the
+results to other `TestCheckFunc` methods. The below example uses
+`ComposeTestCheckFunc` to group a set of `TestCheckFunc` functions together. The
+first function `testAccCheckExampleWidgetExists` uses the `Example` service SDK
+directly, and queries it for the ID of the widget we have in state. After found,
```

Comment:
"After found" feels awkward to me, maybe try "Once found". Not a dealbreaker obviously it just stood out a little.